src/orquestra/sdk/_base/_driver/_exceptions.py

```diff
@@ -52,16 +52,45 @@ def __init__(self, workflow_def_id: str):
 class InvalidWorkflowRunID(Exception):
     """
     Raised when an invalid Workflow Run ID is sent to the Workflow Driver
-
-    Workflow Run IDs are expected to be UUIDs, if an ID is sent that cannot be
-    parsed as a UUID, this exception will be raised.
     """
 
     def __init__(self, workflow_run_id: str):
         self.workflow_run_id = workflow_run_id
         super().__init__(workflow_run_id)
 
 
+class InvalidWorkflowRunArtifactID(Exception):
+    """
+    Raised when an invalid Workflow Run Artifact ID is sent to the Workflow Driver
+
+    Workflow Run Artifact IDs are expected to be UUIDs, if an ID is sent that cannot be
+    parsed as a UUID, this exception will be raised.
+    """
+
+    def __init__(self, workflow_run_artifact_id: str):
+        self.workflow_run_artifact_id = workflow_run_artifact_id
+        super().__init__(workflow_run_artifact_id)
+
+
+class InvalidWorkflowRunResultID(Exception):
+    """
+    Raised when an invalid Workflow Run Result ID is sent to the Workflow Driver
+
+    Workflow Run Result IDs are expected to be UUIDs, if an ID is sent that cannot be
+    parsed as a UUID, this exception will be raised.
+    """
+
+    def __init__(self, workflow_run_result_id: str):
+        self.workflow_run_result_id = workflow_run_result_id
+        super().__init__(workflow_run_result_id)
+
+
+class ForbiddenError(Exception):
```

Comment:
There's also `UnauthorizedError`. Do we need to differentiate between the two cases on the user-facing layer?

Reply:
`UnauthorizedError` is on the user facing level and `ForbiddenError` is an internal exception. On the next layer up `InvalidTokenError` and `ForbiddenError` are wrapped together under `UnauthorizedError`, I don't see an immediate need to have separate exception on the user level.

Reply:
Ohhh right, nevermind. I thought this was `orquestra.sdk.exceptions` but it's not.